rtrlib/aspa/aspa.c

```diff
@@ -308,14 +366,29 @@ static enum aspa_status aspa_table_update_internal(struct aspa_table *aspa_table
 		if (op_type == ASPA_ADD) {
 #ifdef ASPA_UPDATE_IN_PLACE
 			if (aspa_array_insert(array, existing_i, &current->record) != ASPA_SUCCESS) {
+				{
 #else
-			if (aspa_array_append(new_array, &current->record) != ASPA_SUCCESS) {
+			if (lastadded_record && lastadded_record->customer_asn == current->record.customer_asn) {
```

Comment:
The update function should only access the current and the next op. Besides naming, the last record shouldn't be accessed unless absolutely necessary -- why need `last_asn` and `current_withdrawn` then?

Reply:
when not in-place, we may need to overwrite the last-added record (which is in a different data structure than e.g. existing_record) when for a certain customer_asn, multiple announcements and withdrawals take place in one cache update.

Reply:
please see my comment https://github.com/tanneberger/rtrlib/pull/11#issuecomment-1885059021 below